aten/src/ATen/native/cuda/BinaryMiscOpsKernels.cu

```diff
@@ -19,11 +19,11 @@ void atan2_kernel_cuda(TensorIterator& iter) {
   });
 }
 
-void smooth_l1_kernel_cuda(TensorIterator& iter) {
+void smooth_l1_kernel_cuda(TensorIterator& iter, double beta) {
   AT_DISPATCH_FLOATING_TYPES_AND_HALF(iter.dtype(), "smooth_l1_cuda", [&]() {
     gpu_kernel(iter, [] GPU_LAMBDA (scalar_t a, scalar_t b) -> scalar_t {
       auto z = ::abs(a - b);
-      return z < scalar_t(1.) ? scalar_t(0.5) * z * z : z - scalar_t(0.5);
+      return z < beta ? scalar_t(0.5) * z * z / beta : z - scalar_t(0.5) * beta;
```

Comment:
similar here, isn't the math in double now?

Reply:
In general, do we want scalar inputs to our kernels to have the same precision as the dtype of the tensor(s) they operate on? In that case it sounds a little harder since we don't want beta to take on a complex type:
- When scalar_t is a complex, use beta with double precision
- Otherwise, use beta as a scalar_t

Or would another option be just to directly make beta a float rather than a double (which I think would require reducing precision when we go from python to C++).